app/controllers/quotes_controller.rb

```diff
@@ -47,7 +47,11 @@ def destroy
 
   private
 
+  def quoted_person
+    Person.find_or_create_by(full_name: params.require(:quote).permit(:quoted_person)[:quoted_person])
+  end
+
   def quote_params
-    params.require(:quote).permit(:body, :quoted_person, :location)
+    params.require(:quote).permit(:body, :location).merge(person: quoted_person)
```

Comment:
This gets pretty ugly as its another params.require inside of this quote_params.

I'd probably try and do all the nested requires in a single ```.require.permit```, and then pull out afterward.

It'll be easier to understand and audit for param usage in the future.